### PR TITLE
[Button] Deprecate connectedDisclosure 

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris/issues/10126

### CODE REVIEW

File: .changeset/famous-stingrays-change.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Deprecate connectedDisclosure prop on button
```
Comment (sam-b-rose) [inline .changeset/famous-stingrays-change.md:5]:
```suggestion
Deprecated `connectedDisclosure` prop on Button
```

**Review by sam-b-rose — APPROVED** [review]
Very nice! Great composition example in storybook and the docs! So much better 👍

File: polaris-react/src/components/Button/Button.stories.tsx [hunk 2]
```diff
@@ -717,17 +720,31 @@ export function SelectDisclosure() {
...
-      </Button>
+      <ButtonGroup segmented>
+        <Button>Save</Button>
+
+        <Popover
+          active={active}
+          preferredAlignment="right"
+          activator={
+            <Button
+              onClick={() => setActive(true)}
```
Comment (aveline) [inline polaris-react/src/components/Button/Button.stories.tsx:761]: Is it worth keeping the `primary` prop here to avoid a Chromatic diff?

Comment (kyledurand) [reply polaris-react/src/components/Button/Button.stories.tsx:761]: Sure, I can add it 👍

**Review by aveline — APPROVED** [review]
Great to see this accomplished with composability rather than configuration 🙌

**Pull request merged**